framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -150,13 +150,15 @@ protected V getValueFromFactory(Tree tree, Node node) {
             // get the declared type of the assignment context by looking up the
             // assignment context tree's type in the factory while flow is
             // disabled.
-            // Note: Since we use getAnnotatedType(Element), flow is not used in
-            // any case, so we would not have to disable it.
-            boolean oldFlow = factory.getUseFlow();
-            factory.setUseFlow(false);
-            Element element = node.getAssignmentContext().getElementForType();
-            if (element != null) {
-                AnnotatedTypeMirror assCtxt = factory.getAnnotatedType(element);
+            Tree contextTree = node.getAssignmentContext().getContextTree();
+            AnnotatedTypeMirror assCtxt = null;
+            if (contextTree != null) {
+                assCtxt = factory.getAnnotatedTypeLhs(contextTree);
+            } else if( node.getAssignmentContext().getElementForType() != null) {
```

Comment:
getElementForType https://github.com/typetools/checker-framework/blob/master/dataflow/src/org/checkerframework/dataflow/cfg/node/AssignmentContext.java#L40 doesn't look particularly cheap. I would compute it once and put it in a local, instead of computing it twice.